File: PhilterPolicyEditor/Classes/Conditional.vb
Public Class Conditional

    Public Property Expression As String
    Public Property ConditionalOperator As String
    Public Property Value As String

    Public Sub New(conditional As String)

        ' TODO: Parse the string into a conditional.

    End Sub

    Public Sub New(expression As String, conditionalOperator As String, value As String)

        Me.Expression = expression
        Me.ConditionalOperator = conditionalOperator
        Me.Value = value

    End Sub

    Public Function GetConditionalString() As String

        If String.IsNullOrEmpty(Expression) Or String.IsNullOrWhiteSpace(Expression) Then
            Return "no conditional"
        Else
            Return Expression & " " & ConditionalOperator & " " & Chr(34) & Value & Chr(34)
        End If

    End Function

    Public Overrides Function ToString() As String

        Return GetConditionalString()

    End Function

End Class

File: PhilterDesktop/PolicyEditing/Conditional.cs
namespace PhilterDesktop.PolicyEditing
{
    /// <summary>
    /// Helper for building a filter-strategy condition expression string.
    /// Ported from the former VB editor's Conditional class.
    /// </summary>
    internal sealed class Conditional
    {
        public string? Expression { get; set; }
        public string? ConditionalOperator { get; set; }
        public string? Value { get; set; }

        public Conditional(string expression, string conditionalOperator, string value)
        {
            Expression = expression;
            ConditionalOperator = conditionalOperator;
            Value = value;
        }

        public string GetConditionalString()
        {
            return string.IsNullOrWhiteSpace(Expression)
                ? string.Empty
                : $"{Expression} {ConditionalOperator} \"{Value}\"";
        }

        public override string ToString() => GetConditionalString();
    }
}
